p2p/transport/webtransport/transport_test.go

```diff
@@ -178,7 +178,7 @@ func TestHashVerification(t *testing.T) {
 		var trErr *quic.TransportError
 		require.ErrorAs(t, err, &trErr)
 		require.Equal(t, quic.TransportErrorCode(0x12a), trErr.ErrorCode)
-		require.Contains(t, trErr.ErrorMessage, "cert hash not found")
+		require.Contains(t, errors.Unwrap(trErr).Error(), "cert hash not found")
```

Comment:
That API seems a bit awkward. Maybe quic-go should expose an `Error error` field on the `TransportError`? Any thoughs?

Reply:
As long as we set the `error` field on `quic.TransportError` correctly, this seems fine. Any one who needs an exact check can do `errors.Is(err, targetErr)`. Anyone who needs to check string contains can do `strings.Contains(err.Error(), "error string")`

We can also change this test to:
```
require.Contains(t, errors.Error(), "cert hash not found")
```

Reply:
I see the [problem](https://github.com/quic-go/quic-go/blob/master/internal/qerr/errors.go#L39), the `error` field on `TransportError` is only included in the Error() string if `ErrorMessage` is empty. 

Can we include both `ErrorMessage` and `error.Error()` in the final Error() string?

Reply:
In principle yes, but there's a lot going on in `Error` already: https://github.com/quic-go/quic-go/blob/824fd8a2f2eb8a08fe6cef7a693fee6be3819e01/internal/qerr/errors.go#L33-L49

Suggestions welcome!